web/src/hooks/useTtsPlayer.ts

```diff
@@ -0,0 +1,72 @@
+import { useContext, useEffect, useMemo } from 'react'
+import { useLocation } from 'react-router-dom'
+import segment from 'sentencex'
+
+import { parseHTML } from 'shared'
+import { EventModel, LocalNewsModel, PageModel, TunewsModel } from 'shared/api'
+
+import { TtsContext, TtsContextType } from '../contexts/TtsContextProvider'
+
+const useTtsPlayer = (
+  languageCode?: string,
+  model?: PageModel | LocalNewsModel | TunewsModel | EventModel | null,
+): TtsContextType => {
+  const tts = useContext(TtsContext)
+  const location = useLocation()
+
+  const sentences = useMemo(() => {
+    if (model) {
+      const tempDiv = document.createElement('div')
+      tempDiv.innerHTML = model.content
+      const appendPeriod = (elements: NodeListOf<HTMLElement>) => {
+        elements.forEach((element: HTMLElement) => {
+          const trimmedText = element.textContent?.trim()
+          if (trimmedText && !trimmedText.endsWith('.')) {
+            // eslint-disable-next-line no-param-reassign
+            element.textContent = trimmedText.concat('.')
+          }
+        })
+      }
+
+      const listItems = tempDiv.querySelectorAll('li')
+      appendPeriod(listItems)
+
+      const paragraphs = tempDiv.querySelectorAll('p')
+      appendPeriod(paragraphs)
+
+      const textContent = tempDiv.textContent || tempDiv.innerText
+      const content = parseHTML(textContent)
+      return [model.title.concat('.'), ...segment(languageCode, content)]
+    }
+
+    return []
+  }, [model, languageCode])
```

Comment:
💭 Hmmm, I don't like this hack to break up the content, and I see that you didn't do it in the native one. Maybe there is a better solution here?

Reply:
hmm at native it stops normal at the end of sentences but at web I noticed lists and paragraphs without full stop will continue reading the next sentence as it one.

Other option is maybe changes from CMS ?